src/storage-control/src/client.rs

```diff
@@ -261,6 +261,112 @@ impl Storage {
+    /// * `resource` - the bucket name. In `projects/_/buckets/{bucket_id}`
+    ///   format.
+    ///
+    /// # Example
+    /// ```
+    /// # use google_cloud_storage_control::client::Storage;
+    /// async fn example(client: &Storage) -> gax::Result<()> {
+    ///     let policy = client
+    ///         .get_iam_policy("projects/_/buckets/my-bucket")
+    ///         .send()
+    ///         .await?;
+    ///     println!("policy details={policy:?}");
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn get_iam_policy(
+        &self,
+        resource: impl Into<String>,
+    ) -> super::builder::storage::GetIamPolicy {
+        self.inner.get_iam_policy().set_resource(resource.into())
+    }
+
+    /// Updates the IAM policy for a specified bucket.
+    ///
+    /// This is not an update. The supplied policy will overwrite any existing
+    /// IAM Policy. You should first get the current IAM policy with
+    /// `get_iam_policy()` and then modify that policy before supplying it to
+    /// `set_iam_policy()`.
+    ///
+    /// # Parameters
+    /// * `resource` - the bucket name. In `projects/_/buckets/{bucket_id}`
+    ///   format.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use google_cloud_storage_control::client::Storage;
+    /// async fn example(client: &Storage) -> gax::Result<()> {
+    ///     // Get the current IAM policy for this bucket
+    ///     let policy = client
+    ///         .get_iam_policy("projects/_/buckets/my-bucket")
+    ///         .send()
+    ///         .await?;
+    ///
+    ///     // Add a binding to the policy
+    ///     let mut new_policy = policy.clone();
+    ///     new_policy.bindings.push(
+    ///         iam_v1::model::Binding::new()
+    ///             .set_role("roles/storage.legacyBucketReader")
+    ///             .set_members(["user:me@example.com"]),
+    ///     );
+    ///
+    ///     // Update the IAM policy for this bucket
+    ///     let response = client
+    ///         .set_iam_policy("projects/_/buckets/my-bucket")
+    ///         .set_update_mask(wkt::FieldMask::default().set_paths(["bindings"]))
+    ///         .set_policy(new_policy)
+    ///         .send()
+    ///         .await?;
+    ///     println!("response details={response:?}");
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn set_iam_policy(
+        &self,
+        resource: impl Into<String>,
+    ) -> super::builder::storage::SetIamPolicy {
+        self.inner.set_iam_policy().set_resource(resource.into())
+    }
+
+    /// Tests a set of permissions on the given bucket, object, or managed folder
+    /// to see which, if any, are held by the caller.
+    ///
+    /// # Parameters
+    /// * `resource` should be
+    ///   * `projects/_/buckets/{bucket}` for a bucket,
+    ///   * `projects/_/buckets/{bucket}/objects/{object}` for an object, or
+    ///   * `projects/_/buckets/{bucket}/managedFolders/{managedFolder}` for a
+    ///     managed folder.
```

Comment:
It is odd that the available resource names do not match between these functions.  @benwhitehead any thoughts?

Reply:
+1

Reply:
Logged an internal bug (b/414625980) for determination of whether these can also be used for {Get,Set}IamPolicy.

Reply:
Looks like these resource names apply to all the IAM functions . We should add them to the hand-crafted comments.

Reply:
Oops, missed this comment before merging. I will send a follow up with the changes.